classes/plugins/Plugin.inc.php

```diff
@@ -318,6 +344,34 @@ function getTemplatePath($inCore = false) {
 	}
 
 	/**
+	 * Call this method when an enabled plugin is registered in order to override
+	 * template files in other plugins. Any plugin which calls this method can
+	 * override template files by adding their own templates to:
+	 * <overridingPlugin>/templates/plugins/<category>/<originalPlugin>/<path>.tpl
+	 *
+	 * @param $hookName string TemplateResource::getFilename
+	 * @param $args array [
+	 *		@option string File path to preferred template. Leave as-is to not
+	 *			override template.
+	 *		@option string Template file requested
+	 * ]
+	 * @return string New template file location
+	 */
+	public function _overridePluginTemplates($hookName, $args) {
```

Comment:
Be careful to choose your return values for hook callback functions carefully -- a `true` return will stop processing the hook (cutting off later registrants from getting invoked), and a `false` will continue processing. I think a `false` return would be appropriate both for the current `return;` and at the end of the function.

Reply:
Ah, good to know. Does it need to explicitely `false` or is falsey good enough?

Reply:
Falsey should work, but I think we should encourage good habits here by making it explicit. Returning the wrong thing from a plugin hook will generally not cause *your* plugin to fail, but *the next* hook registrant, which is a huge pain to debug.